src/main/java/com/google/songgame/servlets/StartGameServlet.java

```diff
@@ -0,0 +1,38 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.pusher.rest.Pusher;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/start-game")
+public final class StartGameServlet extends HttpServlet {
```

Comment:
I'm wondering if we need a new servlet to handle this specific action.

I'm not familiar with the best way to handle this honestly. Since servlets dont seem to be very easy to use with custom url parameters.

But.. that being said, I imagine there will be a lot of actions that can happen relating to a game, and I'm not sure all of them need a unique servlet. For example, after the start game, there will be the start round right?

perhaps we can make us of the `doPut` method override.

We can then have that take different action depending on a parameter you supply, likely gameId to refer to a specific game, and then an action like "start-game" or "start-round" or "end-game"

a Put request is used to update a resource (in this instance, we are referring to a single game).

Traditionally, a "PATCH" request would be used for this, but the servlets don't support them natively based on my googling. but a PUT request is an adequate replacement.

you can then call fetch on the client but with method: "PUT"

Let me know if this doesn't make sense, or if you have an argument for keeping this servlet here as is.